redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/Node.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood
+
+/**
+ * An interface that has the common behavior as a node contained in a flexbox.
+ */
+public interface Node {
+
+  /**
+   * The width attribute of the node.
+   *
+   * The attribute is about how wide the view wants to be. Can be one of the
+   * constants [MatchParent] or [WrapContent], or an exact size.
+   */
+  public val width: Int
+
+  /**
+   * The height attribute of the node.
+   *
+   * The attribute is about how wide the view wants to be. Can be one of the
+   * constants [MatchParent] or [WrapContent], or an exact size.
+   */
+  public val height: Int
+
+  /**
+   * The minimum width attribute of the node.
+   *
+   * The attribute determines the minimum width the child can shrink to.
+   */
+  public val minWidth: Int
+
+  /**
+   * The minimum height attribute of the node.
+   *
+   * The attribute determines the minimum height the child can shrink to.
+   */
+  public val minHeight: Int
+
+  /**
+   * The maximum width attribute of the node.
+   *
+   * The attribute determines the maximum width the child can expand to.
+   */
+  public val maxWidth: Int
+
+  /**
+   * The maximum height attribute of the node.
+   */
+  public val maxHeight: Int
+
+  /**
+   * True if this item is visible and should be laid out.
+   */
+  public val visible: Boolean
+
+  /**
+   * The baseline used for [AlignItems.Baseline] and [AlignSelf.Baseline].
+   * If not specified, [Undefined] is set as a default value.
+   */
+  public val baseline: Int
+
+  /**
+   * The order attribute of the node.
+   *
+   * The attribute can change the order in which the children are laid out.
+   * By default, children are displayed and laid out in the same order as they added to the
+   * [FlexboxEngine]. If not specified, [DefaultOrder] is set as a default value.
+   */
+  public val order: Int
+
+  /**
+   * The flex grow attribute of the node.
+   *
+   * The attribute determines how much this child will grow if positive free space is
+   * distributed relative to the rest of other nodes included in the same flex line.
+   * If not specified, [DefaultFlexGrow] is set as a default value.
```

Comment:
Is it better to say 0f here?

Reply:
It’s also weird for an interface to specify its default value... this feels like a class to me.

Reply:
...Do users ever compare anything against `DefaultFlexGrow` ? Or is it just a value they use when creating a new implementation?